CountingBloomFilter.hpp

```diff
@@ -88,21 +76,21 @@ class CountingBloomFilter
 	// but never change the type or delete the field.
 	struct FileHeader
 	{
-		char magic[8];
+		char magic[sizeof(uint64_t)];
```

Comment:
```suggestion
		char magic[8];
```
I preferred the explicit `8` in this case. Alternatively, use a `static const unsigned MAGIC_LENGTH = 8;`

Reply:
So the reason I changed it in the first place was because clang-tidy said 8 was a magic number and the solution I came up with was that we choose 8 because it is the sizeof(uint64_t). 

 I also get an error `error: 'MAGIC_LENGTH' was not declared in this scope char magic[MAGIC_LENGTH];` When I try to declare that as a static variable and use it to create the char array.

Reply:
This quick test worked for me:
```c++
class Hello {
	static const unsigned MAGIC_LENGTH = 8;
	char m_magic[MAGIC_LENGTH];
};
```

Reply:
The reason for the error I posted above is because the public FileHeader struct is defined before the private member initialization. I believe that there is no reason for the FileHeader to be public so I've moved it to private and then incorporated your changes.